web/api/v1/api_test.go

```diff
@@ -1012,10 +1013,188 @@ func TestReadEndpoint(t *testing.T) {
 				Samples: []prompb.Sample{{Value: 1, Timestamp: 0}},
 			},
 		},
+	}, resp.Results[0])
+}
+
+func TestStreamReadEndpoint(t *testing.T) {
+	// 3 series each series.
+	// First with 119 samples. We expect 1 frame with 1 chunk.
+	// Second with 121 samples, We expect 1 frame with 2 chunks.
+	// Third with 241 samples. We expect 1 frame with 2 chunks, and 1 frame with 1 chunk for the same series due to bytes limit.
+	suite, err := promql.NewTest(t, `
+		load 1m
+			test_metric1{foo="bar1",baz="qux"} 0+100x119
+            test_metric1{foo="bar2",baz="qux"} 0+100x120
```

Comment:
Comment says `// Second with 121 samples, We expect 1 frame with 2 chunks.` but here we have 120 why?

```suggestion
            test_metric1{foo="bar2",baz="qux"} 0+100x121
```

Reply:
so `0 + X` means X+1 samples (: 

Fixed `119 samples` comment as there are 120 exactly.

Reply:
yep that was the confusion, just the comment than, thanks